File: src/main/resources/statics/js/modules/longcoin/longcoinlocalrecord.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'longcoin/longcoinlocalrecord/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '设备号', name: 'sid', index: 'sid', width: 80 },
			{ label: '流水号', name: 'seq', index: 'seq', width: 80 }, 			
			{ label: '用户账户', name: 'useraccount', index: 'useraccount', width: 80 }, 			
            { label: '账户类型', name: 'accounttype', width: 60, formatter: function(value, options, row){
                            return value =='1' ?'<span class="label label-success">手机号</span>':(value =='2' ?'<span class="label label-success">微信openId</span>':
                                (value =='3' ?'<span class="label label-success">微信unionId</span>':(value =='4' ?'<span class="label label-success">未知</span>':'其他')));
                        }},
			 { label: '操作类别', name: 'operatortype', width: 60, formatter: function(value, options, row){
                                        return value =='2' ?'<span class="label label-success">充值</span>':(value =='3' ?'<span class="label label-success">消费</span>':'其他');

                                    }},
			{ label: '充值消耗类别ID ', name: 'operatortypeid', index: 'operatortypeid', width: 80 }, 			
			{ label: '金额', name: 'amount', index: 'amount', width: 80 }, 			
			{ label: '本地创建时间', name: 'createTime', index: 'create_time', width: 80 }, 			
			{ label: '是否已对账', name: 'checked', width: 60, formatter: function(value, options, row){
                return value =='0' ?'<span class="label label-success">否</span>':(value =='1' ?'<span class="label label-success">已对账</span>':'其他');

            }},
			{ label: '对账结果', name: 'checkedResult', width: 60, formatter: function(value, options, row){
                return value =='0' ?'<span class="label label-success">本地多</span>':(value =='1' ?'<span class="label label-success">平账</span>':
                (value =='2' ?'<span class="label label-success">龙币商城多</span>':'其他'));
            }},
			{ label: '对账时间', name: 'checkedTime', index: 'checked_time', width: 80 }, 			
			{ label: '异常处理状态', name: 'exceptionFlag', width: 60, formatter: function(value, options, row){
                            return value =='0' ?'<span class="label label-success">未处理</span>':(value =='1' ?'<span class="label label-success">正常</span>':'已处理');

                        }},
			{ label: '处理结果', name: 'result', index: 'result', width: 80 },
			{ label: '来源渠道', name: 'fromType', width: 60, formatter: function(value, options, row){
                            return value =='1' ?'<span class="label label-success">百色现场会</span>':(value =='2' ?'<span class="label label-success">佛山推多多</span>':
                            (value =='3' ?'<span class="label label-success">其他</span>':'未知'));
                        }},
			{ label: '订单状态', name: 'status', width: 60, formatter: function(value, options, row){
                                        return value =='0' ?'<span class="label label-success">创建</span>':(value =='1' ?'<span class="label label-success">成功</span>':
                                        (value =='2' ?'<span class="label label-success">失败</span>':'未知'));
                                    }},
			{ label: '消费充值结果描述', name: 'statusResult', index: 'status_result', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		longcoinLocalrecord: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.longcoinLocalrecord = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.longcoinLocalrecord.id == null ? "longcoin/longcoinlocalrecord/save" : "longcoin/longcoinlocalrecord/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.longcoinLocalrecord),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "longcoin/longcoinlocalrecord/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "longcoin/longcoinlocalrecord/info/"+id, function(r){
                vm.longcoinLocalrecord = r.longcoinLocalrecord;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});

File: src/main/java/com/icloud/modules/longcoin/entity/LongcoinLocalrecord.java
package com.icloud.modules.longcoin.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 本地龙币消费记录
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-11-12 16:32:46
 */
@Data
@TableName("t_longcoin_localrecord")
public class LongcoinLocalrecord implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /*  */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /* 每个服务器分配的消费龙币的设备号 */
       @TableField("sid")
       private String sid;
   	   	   /* 流水号 */
       @TableField("seq")
       private String seq;
   	   	   /* 用户账户 */
       @TableField("useraccount")
       private String useraccount;
   	   	   /* 账户类型（1、手机号码 2、微信openId 3、微信unionId） */
       @TableField("accounttype")
       private String accounttype;
   	   	   /* 操作类别 (2、充值、3消费) */
       @TableField("operatortype")
       private String operatortype;
   	   	   /* 充值消耗类别ID  */
       @TableField("operatortypeid")
       private String operatortypeid;
   	   	   /* 金额 */
       @TableField("amount")
       private Integer amount;
   	   	   /* 本地创建时间 */
       @TableField("create_time")
       private Date createTime;
   	   	   /* 是否已对账 0否 1已对账 */
       @TableField("checked")
       private String checked;
   	   	   /* 账结果1平0本地多2龙币商城多 */
       @TableField("checked_result")
       private String checkedResult;
   	   	   /* 对账时间 */
       @TableField("checked_time")
       private Date checkedTime;
   	   	   /* 异常处理状态1正常，0未处理2已处理 */
       @TableField("exception_flag")
       private String exceptionFlag;
   	   	   /* 处理结果 */
       @TableField("result")
       private String result;
   	   	   /* 支付状态 0未支付 1已支付 2关闭 */
       @TableField("pay_status")
       private String payStatus;
   	   	   /* 支付状态 0未退款 1已退款 2关闭 */
       @TableField("refund_status")
       private String refundStatus;
   	   	   /* 来源渠道 1、百色现场会 2、 */
       @TableField("from_type")
       private String fromType;
   	   	   /* 0创建、1成功、2失败 */
       @TableField("status")
       private String status;
   	   	   /* 消费充值结果描述 */
       @TableField("status_result")
       private String statusResult;
   	
}
